awx/main/analytics/collectors.py

```diff
@@ -282,32 +282,34 @@ def _copy_table(table, query, path):
 
 @register('events_table', '1.2', format='csv', description=_('Automation task records'), expensive=True)
 def events_table(since, full_path, until, **kwargs):
-    events_query = '''COPY (SELECT main_jobevent.id, 
+
+    safe_event_data = "replace(main_jobevent.event_data::text, '\\u0000', '')::json"
```

Comment:
@chrismeyersfsu @Ladas is there any additional performance cost on doing a `replace()` on every row?

I'd be curious to see what an explain and timing looks like for a medium sized table with actual data here.

If there's an added cost, perhaps we can do what I suggested @chrismeyersfsu:

```
try:
    return _copy_table(...)
except ThatError as e:
    if e.errno == 'whatever':
        return _copy_table(query_with_replace)
```

Reply:
Would the billion event workload be good for testing this?

Reply:
I doubt you even need that many.

Reply:
```sql
awx=> \timing
awx=> SELECT CAST(replace(main_jobevent.event_data::text, '\\u0000', '')::json->>'start' AS TIMESTAMP WITH TIME ZONE) AS start FROM main_jobevent LIMIT 200000;
Time: 6738.434 ms (00:06.738)
awx=> SELECT CAST(main_jobevent.event_data::json->>'start' AS TIMESTAMP WITH TIME ZONE) AS start FROM main_jobevent LIMIT 200000;
Time: 5823.323 ms (00:05.823)
```

Reply:
so 15% increase and it might vary based on various sizes of the json I guess

That should be ok as sanity check perf-test (we should be loading a batch of events always, so hopefully not 1B per batch, although that can happen with the time based chunking)

Reply:
I'm guessing how much it increases varies widely based on the size of `event_data`; for really large event data, the cost of a full text search could be pretty expensive, I think.

Reply:
@ryanpetrello   well, we could try to catch a unicode failure and then try it again with this code?

Reply:
Is the performance gain worth the added complexity? Specifically, when it comes to performance testing. Does this add another dimension that will always need to be verified or do we just sort of know "ok, that replace adds a 15% overhead"

Reply:
I'd like to believe that this is an outlier, in that case what @ryanpetrello says would help (we know we have some large json blobs, like playbook on stats, so the extra processing time can be much bigger)

But we don't have any metrics about the event collection failures at customer side, so it's just a guess. If this would appear very often, we'd make the collection much slower (depending how fast it'll fail first time, so up to like ~2x slower).

Reply:
`http://host/api/v2/job_events/?event_data__regex=\\u0000` *WARNING* This query will be EXPENSIVE

Reply:
@chrismeyersfsu IT has a looots of events, the API query till probably timeout? We should be able to limit this to e.g. last few days?

Reply:
Good idea on the date restriction @Ladas . https://tower.ansible.eng.rdu2.redhat.com/api/v2/jobs/?created__gte=2021-02-24T00:00:00.00Z&job_events__event_data__contains=%5C%5Cu0000

Salt the above to taste. Currently the date is set to jobs > `February 24, 2021`